modules/040-node-manager/hooks/update_approval.go

```diff
@@ -69,6 +71,33 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 			},
 			FilterFunc: updateApprovalFilterNode,
 		},
+		{
+			Name:       "deckhouse_pod_node",
+			ApiVersion: "v1",
+			Kind:       "Pod",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-system"},
+				},
+			},
+			LabelSelector: &v1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "deckhouse",
+				},
+			},
+			FieldSelector: &types.FieldSelector{
+				MatchExpressions: []types.FieldSelectorRequirement{
+					{
+						Field:    "status.phase",
+						Operator: "Equals",
+						Value:    "Running",
+					},
+				},
+			},
+			ExecuteHookOnEvents:          pointer.BoolPtr(false),
+			ExecuteHookOnSynchronization: pointer.BoolPtr(false),
+			FilterFunc:                   filterDeckhousePodNode,
```

Comment:
It is possible to use downward API to get the deckhouse pod node and avoid Kubernetes API interactions:
```yaml
env:
- name: MY_NODE_NAME
  valueFrom:
    fieldRef:
      fieldPath: spec.nodeName
```

Reply:
I chosen Maksim's variant because we manage env variable and HOSTNAME variable can changed, for example, by new containerd, k8s... etc behavior.